### PR TITLE
Output file option

### PR DESCRIPTION
**Pull request by BinderDavid**
Enable use of "-o" option.

### CODE REVIEW

**Comment by BinderDavid** [comment]
Resolves #70

**Comment by BinderDavid** [comment]
@mpickering For some reason CI died with an out-of-memory error.

File: main/Main.hs [hunk 1]
```diff
@@ -30,17 +31,30 @@ argsToOutput a =
...
+doOneJson a fin fout = do
+  (_, val) <- generateJson fin a
+  encodeFile fout val
+
 doJson :: Args -> IO ()
-doJson a = do
-  forM_ (files a) $ \file -> do
-    (_, val) <- generateJson file a
-    encodeFile (file <.> "json") val
+doJson a@((files &&& outputFile) -> (fs, Nothing)) =
```
Comment (mpickering) [inline main/Main.hs]: It is more idiomatic to say:

```
doJson a@Args{files = files, outputFile = Nothing} =
```

File: main/Main.hs [hunk 2]
```diff
@@ -30,17 +31,30 @@ argsToOutput a =
...
-doHtml a = do
-  forM_ (files a) $ \file -> do
-    (header, data_json) <- generateJson file a
-    let html = templateString header data_json a
-    let filename2 = file <.> "html"
-    writeFile filename2 html
-    exitSuccess
+doHtml a@((files &&& outputFile) -> (fs, Nothing)) =
+  forM_ fs $ \file -> doOneHtml a file (file <.> "html")
+doHtml a@((files &&& outputFile) -> ([fin], Just fout)) =
```
Comment (mpickering) [inline main/Main.hs]: It's a shame to duplicate this logic in two places. Is there a way to remove the duplication?

**Comment by mpickering** [comment]
Thanks. Can you please add a comment to the changelog as well. I'm "fixing" CI, and will try to make a more permanent fix shortly.

**Comment by mpickering** [comment]
I fixed CI now by complaining to the right people.

File: main/Main.hs (hunk 1 above)
Comment (BinderDavid) [reply main/Main.hs]: Thanks, I applied the change.

File: main/Main.hs (hunk 2 above)
Comment (BinderDavid) [reply main/Main.hs]: I refactored the logic so the duplication is gone now.

**Pull request merged**